lldb-eval/eval.cc

```diff
@@ -781,6 +817,120 @@ Value Interpreter::EvaluateBinaryBitwise(BinaryOpKind kind, Value lhs,
   return EvaluateArithmeticOpInteger(target_, kind, lhs, rhs, lhs.type());
 }
 
+Value Interpreter::EvaluateAssignment(Value lhs, Value rhs) {
+  assert(CompareTypes(lhs.type(), rhs.type()) &&
+         "invalid ast: operands must have the same type");
+
+  lhs.Update(rhs);
+  return lhs;
+}
+
+Value Interpreter::EvaluateBinaryAddAssign(Value lhs, Value rhs) {
+  Value ret;
+
+  if (lhs.IsPointer()) {
+    assert(rhs.IsInteger() && "invalid ast: rhs must be an integer");
+    ret = EvaluateBinaryAddition(lhs, rhs);
+  } else {
+    assert(lhs.IsScalar() && "invalid ast: lhs must be an arithmetic type");
+    assert(rhs.type().IsBasicType() && "invalid ast: rhs must be a basic type");
```

Comment:
Just curious why is `IsBasicType()` check used here (and not in the parser for example)?
I see there is `lldb::eBasicTypeNullPtr`. Could this cause issues in the future?

Reply:
Parser has all these checks too and if everything works correctly, this assert should never trigger :)

Asserts in `eval.cc` are additional layer validation to catch bugs if something is wrong in the parser. They're disabled in `release` builds, so crashes will happen only in tests.

I guess it would be better to exclude `lldb::eBasicTypeNullPtr` from this check, but there're some other things there too. I think for now it's fine to leave it as it.